### ISSUE TITLE
Partial application of checked functions throws Throwable

### ISSUE THREAD

**Issue by SimY4** [javaslang/javaslang#1343]
Currently all CheckedFunction interfaces has their partial application methods declared with `throws Throwable` but in fact its close to impossible that it'll throw any exception at all. For example:

    @FunctionalInterface
    public interface CheckedFunction2<T1, T2, R> extends λ<R> {
        ...
        CheckedFunction1<T2, R> apply(T1 t1) throws Throwable {
            return (T2 t2) -> apply(t1, t2);
        }
        ...
    }

Basically, all it does is returning new instance of CheckedFunction with reduced arity. This makes currying of checked functions hardly usable since it's necessary to handle `Throwable` each and every time.
In my opinion, since partial function application is not a real application `throws Throwable` should be removed from partial application methods signature. But it'll probably break API compatibility.

**Comment by danieldietrich** [comment]
Fixed with #1259
